cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py

```diff
@@ -0,0 +1,312 @@
+        """Rename the node_properties and resources directories."""
+        base = os.path.join('/opt/cloudify', node_name)
+        properties_dir = os.path.join(base, 'node_properties')
+        properties_dir_backup = os.path.join(base, 'node_properties_backup')
+        resources_dir = os.path.join(base, 'resources')
+        resources_dir_backup = os.path.join(base, 'resources_backup')
+
+        with self.manager_env_fabric() as fabric:
+            fabric.sudo('mv {0} {1}'.format(properties_dir,
+                                            properties_dir_backup))
+            fabric.sudo('mv {0} {1}'.format(resources_dir,
+                                            resources_dir_backup))
+
+        try:
+            yield
+        finally:
+            with self.manager_env_fabric() as fabric:
+                fabric.sudo('mv {0} {1}'.format(properties_dir_backup,
+                                                properties_dir))
+                fabric.sudo('mv {0} {1}'.format(resources_dir_backup,
+                                                resources_dir))
+
+    def test_node_directories(self):
+        """Validation checks that the required directory structure exists.
+
+        Rename the node_properties and resources directories, check that
+        validation breaks when it notices they're missing, then restore their
+        original names.
+        """
+        blueprint_path = self.get_simple_blueprint()
+        inputs = self.get_upgrade_inputs()
+
+        with self.cfy.maintenance_mode():
+            for node_name in [
+                'influxdb',
+                'rabbitmq',
+                'elasticsearch',
+                'amqpinflux',
+                'logstash',
+                'restservice',
+                'nginx',
+                'riemann',
+                'mgmtworker',
+            ]:
+                with self.move_upgrade_dirs(node_name):
+                    try:
+                        self.cfy.upgrade_manager(
+                            blueprint_path=blueprint_path,
+                            inputs_file=inputs,
+                            validate_only=True)
+                    except sh.ErrorReturnCode as e:
+                        self.assertIn(
+                            'service {0} has no properties file'.format(
+                                node_name),
+                            e.stdout.lower())
+                    else:
+                        self.fail('Upgrade directories validation for {0} '
+                                  'succeeded, but it should have failed - '
+                                  'node properties and rollback directories '
+                                  'are missing'.format(node_name))
+
+    def test_rabbit_credentials_changed(self):
+        """Validation doesn't allow changing rabbitmq credentials.
+
+        Try upgrading with a changed rabbitmq password, check that validation
+        breaks.
+        """
```

Comment:
Done in `test_manager_security_settings_changed` and `test_manager_ssh_user_changed`